tests/test_notebooks.py

```diff
@@ -51,7 +46,67 @@ def test_run_all_notebooks():
     proc.wait()
     assert proc.returncode == 0
     # now run
-    cmd = ["python", "build.py",  "--config", "build-ci.yml", "--test"]
+    cmd = ["python", "build.py", "--config", get_build_config(), "--exec"]
     proc = subprocess.Popen(cmd)
     proc.wait()
     assert proc.returncode == 0
+    find_broken_links()
+
+
+@pytest.mark.component
+def test_broken_links():
+    find_broken_links()
+
+
+def find_broken_links(rebuild=True):
+    """Run the Sphinx link checker.
+
+    This was created in response to a number of broken links in Jupyter notebook
+    cells, but would also find broken links in any documentation pages.
+    """
+    os.chdir(_root)
+    config = get_build_config()
+    config_dict = load_build_config(config)
+    # Copy notebooks to docs. -S suppresses Sphinx output.
+    args = ["python", "build.py", "--config", config, "-Sy"]
+    proc = subprocess.Popen(args)
+    rc = proc.wait()
+    assert rc == 0, "Copying notebooks to docs failed"
+    # Run linkchecker (-l). -S suppresses Sphinx output.
+    # output will be in dir configured in sphinx.linkcheck_dir (see below)
+    proc = subprocess.Popen(["python", "build.py", "--config", config, "-Sl"])
+    rc = proc.wait()
+    assert rc == 0, "Linkchecker process failed"
+    # find links marked [broken], report them
+    link_file = Path(".") / config_dict["sphinx"]["linkcheck_dir"] / "output.json"
+    assert link_file.exists()
+    links = []
+    for line in link_file.open(mode="r", encoding="utf-8"):
+        obj = json.loads(line)
+        if obj["status"] == "broken":
+            num = len(links) + 1
+            links.append(f"{num}) {obj['filename']}:{obj['lineno']} -> {obj['uri']}")
+    # fail if there were any broken links
+    assert len(links) == 0, f"{len(links)} broken links:\n" f"{newline.join(links)}"
```

Comment:
I'm fully aware that I'm saying this as a `pytest.fixture` evangelist, but I think that extracting the test setup (i.e. running the command) and the test (i.e. the assertion that the command was successful) would be tidier, not to mention that it would open up the possibility to parametrize some of the tests, e.g. having a `command` fixture that's parametrized with the various CLI flags.

On the other hand, I'm also aware that this is firmly in "testing the tests" territory and so this might be overthinking it, so let's just consider this a note for possible future work if/when we need to make changes to these tests in the future.